src/d3d9/d3d9_device.cpp

```diff
@@ -309,6 +309,14 @@ namespace dxvk {
 
     // When in SWVP mode, 256 matrices can be used for indexed vertex blending
     pCaps->MaxVertexBlendMatrixIndex = m_isSWVP ? 255 : 8;
+    // Only 9Ex devices advertise D3DPBLENDCAPS_SRCCOLOR2 and D3DPBLENDCAPS_INVSRCCOLOR2
+    if (!IsExtended()) {
```

Comment:
It would be cleaner to just conditionally add those two flags if the adapter is extended in Adapter::GetDeviceCaps rather than conditionally removing them in **both** the device and the interface.

Reply:
Done. Agreed it's cleaner this way. I was operating under the false impression it would be a pain to check if a device is extended in the adapter (I stopped asking my brain why it keeps doing these things a long time ago).